core/src/main/resources/org/apache/spark/ui/static/executorspage.js

```diff
@@ -143,23 +143,205 @@ var sumOptionalColumns = [3, 4];
         $("[data-exec-col-idx=" + execOptionalColumns[k] + "]").prop("checked", true);
+        var executorSumCheckBoxId = $("[data-exec-col-idx=" + execOptionalColumns[k] + "]").attr('exec-sum-idx');
+        var selectedExecutorSummaryMetrics = executorSummaryMetricsTableArray.filter(row => (row.executorSumCheckBoxId).toString() == executorSumCheckBoxId)
+        for(var value in selectedExecutorSummaryMetrics) {
+          executorSummaryMetricsTableCurrentStateArray.push(selectedExecutorSummaryMetrics[value]);
+        }
+        executorSummaryMetricsTableCurrentFilteredArray = executorSummaryMetricsTableCurrentStateArray.slice();
       } else {
         allChecked = false;
       }
     }
   }
+
+  createDataTableForExecutorSummaryMetricsTable(executorSummaryMetricsTableCurrentFilteredArray)
+
   if (allChecked) {
     $("#select-all-box").prop("checked", true);
   }
 }
 
+function getColumnNameForExecutorMetricSummary(columnKey) {
+  switch(columnKey) {
+    case "JVMHeapMemory":
+      return "Peak JVM Heap Memory";
+
+    case "JVMOffHeapMemory":
+      return "Peak JVM Off Heap Memory";
+
+    case "OnHeapExecutionMemory":
+      return "Peak On Heap Execution Memory";
+
+    case "OffHeapExecutionMemory":
+      return "Peak Off Heap Execution Memory";
+
+    case "OnHeapStorageMemory":
+      return "Peak On Heap Storage Memory";
+
+    case "OffHeapStorageMemory":
+      return "Peak Off Heap Storage Memory";
+
+    case "OnHeapUnifiedMemory":
+      return "Peak On Heap Unified Memory";
+
+    case "OffHeapUnifiedMemory":
+      return "Peak Off Heap Unified Memory";
+
+    case "DirectPoolMemory":
+      return "Peak Direct Pool Memory";
+
+    case "MappedPoolMemory":
+      return "Peak Mapped Pool Memory";
+
+    case "ProcessTreeJVMVMemory":
+      return "Process Tree JVM VMemory";
+
+    case "ProcessTreeJVMRSSMemory":
+      return "Process Tree JVM RSS Memory";
+
+    case "ProcessTreePythonVMemory":
+      return "Process Tree Python VMemory";
+
+    case "ProcessTreePythonRSSMemory":
+      return "Process Tree Python RSS Memory";
+
+    case "ProcessTreeOtherVMemory":
+      return "Process Tree Other VMemory";
+
+    case "ProcessTreeOtherRSSMemory":
+      return "Process Tree Other RSS Memory";
+
+    case "MinorGCCount":
+      return "Peak Minor GC Count";
+
+    case "MinorGCTime":
+      return "Peak Minor GC Time";
+
+    case "MajorGCCount":
+      return "Peak Major GC Count";
+
+    case "MajorGCTime":
+      return "Peak Major GC Time";
+
+    default:
+      return "NA";
+  }
+}
+
+function createRowMetadataForColumn(colKey, data, executorSumCheckBoxId) {
+  var row = {
+    "metric": getColumnNameForExecutorMetricSummary(colKey),
+    "data": data,
+    "executorSumCheckBoxId": executorSumCheckBoxId,
+    "columnKey": colKey
+  };
+  return row;
+}
+
+function displayRowsForSummaryMetricsTable(row, type, columnIndex) {
+  var str;
+  switch(row.columnKey) {
+    case 'MinorGCCount':
```

Comment:
Why are the GC related keys processed twice? I am seeing them in line 909

Reply:
> Why are the GC related keys processed twice? I am seeing them in line 909

One for data format, one for column meta.